### PR TITLE
`Programming exercises`: Fix an issue where the checkbox to allow complaints on automatic assessment was incorrectly disabled

### PR DESCRIPTION
**Pull request by tobias-lippert**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
@Strohgelaender noticed that the checkbox was disabled in cases where it should be enabled.

### Description
The condition to disable the checkbox was extended with an additional check whether it's an exam exercise or not.
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with complaints enabled

Generate a programming exercise and make sure that the checkbox to allow complaints for automatic assessments is enabled if the assessment mode is automatic and a due date is set and disabled in all other cases.

#### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 1 Exam

1. Generate a programming exercise and make sure that the checkbox to allow complaints for automatic assessments is enabled if the assessment mode is automatic and disabled in all other cases.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2
#### Exam Mode Test
- [x] Test

### CODE REVIEW

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-lifecycle.component.spec.ts [hunk 1]
```diff
@@ -243,4 +245,41 @@ describe('ProgrammingExerciseLifecycleComponent', () => {
...
+        comp.isExamMode = examMode;
+        if (!examMode) {
+            exercise.course = new Course();
+            exercise.course.complaintsEnabled = true;
+        }
+        fixture.detectChanges();
+        const checkbox: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#allowComplaintsForAutomaticAssessment');
+        expectElementToBeDisabled(checkbox);
+    });
+    it('should disable checkbox for complaints on automatic assessments for automatically assessed course exercises without due date', () => {
```
Comment (Strohgelaender) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-lifecycle.component.spec.ts:274]:
```suggestion
    });

    it('should disable checkbox for complaints on automatic assessments for automatically assessed course exercises without due date', () => {
```

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-lifecycle.component.spec.ts [hunk 2]
```diff
@@ -243,4 +245,41 @@ describe('ProgrammingExerciseLifecycleComponent', () => {
         expect(alertSpy).toHaveBeenCalledTimes(nthCall + 1);
         expect(alertSpy).toHaveBeenNthCalledWith(++nthCall, 'artemisApp.programmingExercise.timeline.alertNewExampleSolutionPublicationDateAsDueDate');
     });
+    it.each([true, false])('should enable checkbox for complaints on automatic assessments for automatically assessed exercises', (examMode: boolean) => {
```
Comment (Strohgelaender) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-lifecycle.component.spec.ts:248]:
```suggestion

    it.each([true, false])('should enable checkbox for complaints on automatic assessments for automatically assessed exercises', (examMode: boolean) => {
```

**Review by Strohgelaender — APPROVED** [review]
looks good, thanks for also adding test for this.

**Review by PhilipWerz — CHANGES_REQUESTED** [review]
Button works as expected in both cases (exam and normal).
For Exam Mode color/highlighting is wrong. Even if "Automatic Assesment" is activated text is still grey instaed of black:

Exam Mode (Automatic assesment):
image

Normal PE (Automatic Assesment):
image

Tested on TS1.

**Review by luisa1999 — DISMISSED** [review]
Code looks good to me. No impediments!

**Review by Strohgelaender — DISMISSED** [review]

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-lifecycle.component.spec.ts [hunk 3]
```diff
@@ -243,4 +245,43 @@ describe('ProgrammingExerciseLifecycleComponent', () => {
...
+        if (!examMode) {
+            exercise.course = new Course();
+            exercise.course.complaintsEnabled = true;
+        }
+        fixture.detectChanges();
+        const checkbox: HTMLInputElement = fixture.debugElement.nativeElement.querySelector('#allowComplaintsForAutomaticAssessment');
+        expectElementToBeDisabled(checkbox);
+    });
+
+    it('should disable checkbox for complaints on automatic assessments for automatically assessed course exercises without due date', () => {
```
Comment (JohannesStoehr) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-lifecycle.component.spec.ts:276]: Where is the difference in the set up to the first new test with examMode = false?

Comment (tobias-lippert) [reply src/test/javascript/spec/component/programming-exercise/programming-exercise-lifecycle.component.spec.ts:272]: Thanks for noticing this flaw in my test logic. Should be fixed now.

**Review by PhilipWerz — APPROVED** [review]
Previous remark works now -> Everything works as expected. Tested on Ts1.

**Review by timo07k — APPROVED** [review]
Everything works as expected. Tested on TS1.

**Review by PhilipWerz — APPROVED** [review]
Previous remark works now -> Everything works as expected. Tested on Ts1.

**Review by Strohgelaender — APPROVED** [review]

**Review by JohannesStoehr — APPROVED** [review]
New tests now look good as well

**Pull request merged**